### ISSUE TITLE
GBA not recognizing DPad of controller (using left stick inputs instead)

### ISSUE THREAD

**Issue by CapoFantasma97** [Swordfish90/Lemuroid#592]
**Describe the bug**
I am using my 8BitDo SF30Pro in Switch Pro mode to play GBA games on Lemuroid.
The games register only the stick as directional inputs, and ignores the DPad.
In the rebinding options for controllers, the controller is properly recognized as a Switch Pro device; the analog sticks and the DPad have no entries to set binds.

**Expected behavior**
GBA should recognize the DPad for inputs; being able to use the stick would be nice in some contexts, but not at the expense of not being able to use the DPad.
I don't know if other cores would have the same issue.

**Tablet (please complete the following information):**
 - Device: Samsung Tab S7 Plus
 - OS: Android 13 (OneUI 5.0)

**Comment by yagosaboia** [comment]
The same issue happens to me while using the 8BitDo Bluetooth Ultimate.
I'm able to use the DPad on Lemuroid menu but it doesn't work inside of any game.

Tried with Citra standalone app, and while it doesn't set these binds automatically, it works when you set them manually.

Device: Lenovo P12 Pro
OS: Android 12

**Comment by CapoFantasma97** [comment]
It's also noteworthy that Retroarch does not have this issue.

**Comment by dylanwatsonsoftware** [comment]
Yeah, just to add an extra data point, I have this issue with my PS3 controller cabled to my Pixel 3, playing PSP games (I believe other systems too but will confirm).

**Comment by dylanwatsonsoftware** [comment]
I'd be keen to take a look, if anyone can point me in the right direction to start digging, code-wise.

**Comment by Hapachew** [comment]
To add more information, I also have this problem using my switch pro controller connected over Bluetooth, on a Pixel 5 running Android 13.

**Comment by dylanwatsonsoftware** [comment]
Interestingly I don't have this issue with a Razer Kishi Controller connected to my Android.

**Comment by Redwing321** [comment]
I'm having the same problem but both the dpad and alalog sticks don't work, idk what to do.

**Comment by TimVanDyke** [comment]
On SNES I'm seeing the same thing where left analog stick works but dpad does not. 

Bluetooth connected to Moto G Stylus 5G 2024 - Android 14
 -> Using Bluetooth Auto Connect to force the pro controller to work as an HID and not lag like crazy
      \-> Used version here: https://apkcombo.com/bluetooth-auto-connect/org.myklos.btautoconnect/ (couldn't find documentation for that anywhere, I made ChatGPT spit out a link to install one that let me do this)
